src/lib.rs

```diff
@@ -33,6 +33,8 @@
 // Enable some additional warnings and lints.
 #![warn(clippy::ptr_as_ptr, missing_docs, unused)]
 #![deny(clippy::all)]
+#![deny(rustdoc::all)]
```

Comment:
I think it would actually better to use `warn` instead of `deny` (for clippy too), and treat warnings as errors only in CI. This has the advantage of not getting in the way during local development (you don't necessarily want to fix every warning as you go along, but still be able to run clippy or rustdoc). And we still ensure that all warnings get dealt with in the end by enforcing in CI.

Reply:
That sounds reasonable, I'll drop my objection then :)